backend/src/packages/articles/article.repository.ts

```diff
@@ -14,15 +19,28 @@ class ArticleRepository implements IRepository {
     this.articleModel = articleModel;
   }
 
+  public getArticlesTotal({
+    userId,
+  }: { userId?: number } = {}): Promise<number> {
+    return this.articleModel
+      .query()
+      .where(getWhereUserIdQuery(userId))
+      .resultSize();
+  }
+
   public async findAll({
     userId,
+    take = DEFAULT_PAGINATION_TAKE,
+    skip = DEFAULT_PAGINATION_SKIP,
   }: {
     userId?: number;
-  }): Promise<ArticleEntity[]> {
+  } & ArticlesFilters): Promise<ArticleEntity[]> {
     const articles = await this.articleModel
       .query()
       .where(getWhereUserIdQuery(userId))
       .orderBy('articles.publishedAt', SortingOrder.DESCENDING)
+      .offset(skip)
+      .limit(take)
```

Comment:
And it will return paginated result with `{ results: <rec_typ>[], total: number; }`
```suggestion
      .page(skip, take)
```